cf_xarray/parametric.py

```diff
@@ -0,0 +1,864 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+
+import numpy as np
+import xarray as xr
+from xarray import DataArray
+
+ocean_stdname_map = {
+    "altitude": {
+        "zlev": "altitude",
+        "eta": "sea_surface_height_above_geoid",
+        "depth": "sea_floor_depth_below_geoid",
+    },
+    "height_above_geopotential_datum": {
+        "zlev": "height_above_geopotential_datum",
+        "eta": "sea_surface_height_above_geopotential_datum",
+        "depth": "sea_floor_depth_below_geopotential_datum",
+    },
+    "height_above_reference_ellipsoid": {
+        "zlev": "height_above_reference_ellipsoid",
+        "eta": "sea_surface_height_above_reference_ellipsoid",
+        "depth": "sea_floor_depth_below_reference_ellipsoid",
+    },
+    "height_above_mean_sea_level": {
+        "zlev": "height_above_mean_sea_level",
+        "eta": "sea_surface_height_above_mean_sea_level",
+        "depth": "sea_floor_depth_below_mean_sea_level",
+    },
+}
+
+
+def _derive_ocean_stdname(**kwargs):
```

Comment:
```suggestion
def _derive_ocean_stdname(*, zlev=None, depth=None, eta=None) -> ???
```

This way the language will help you ensure correctness :)